### PR TITLE
[Manual Taxes] Educational Dialog

### LINKED ISSUE #10515: [Manual Taxes] Show the Tax Rate Educational Dialog

### PR DESCRIPTION
**Pull request by toupper**
Closes: #10515 

## Description

With this PR we show the tax educational dialog as part of the Manual Taxes project. 

## Changes
- Refactor `EditableOrderViewModel` so we can generate the `TaxEducationalDialogViewModel` with the right parameters. We pass the tax lines and the "tax based on" setting so they can be displayed accordingly.
- Create the TaxEducationalDialogView
- Create the `TaxEducationalDialogViewModel`
- Display the tax section of wp-admin when the button is tapped.
- Reload the order and the "tax based on" setting whenever the wp-admin web view is closed.
- Unit tests.

## Testing instructions

1. Go to orders
2. Tap on + to create a new order
3. Tap on the ? button on the Taxes Total row.
4. See that we show the tax educational dialog, without the tax lines or "tax based on" setting information.
5. Add products or fees so tax lines are created (make sure you have tax rates in wp-admin/admin.php?page=wc-settings&tab=tax and the products are taxable).
6. Tap again on the ? button on the Taxes Total row.
7. See that the tax lines and "tax based on" setting information is displayed.
8. Tap on the _Edit Tax Rates in Admin_ button. See that you go to the Tax settings in wp-admin. 
9. Close it. See that the order is refreshed so we can update any information related to taxes.

### Educational Dialog with and without tax lines:

https://github.com/woocommerce/woocommerce-ios/assets/1864060/f927aaa3-1866-4f9f-8317-6ffc304b8d68

### The order is refreshed after closing the tax settings in wp-admin:

https://github.com/woocommerce/woocommerce-ios/assets/1864060/784c702f-332c-4489-bd57-9d96ec93cef0

---
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift [hunk 1]
```diff
@@ -153,7 +153,7 @@ final class EditableOrderViewModel: ObservableObject {
 
     /// Defines the tax based on setting to be displayed in the Taxes section.
     /// 
-    @Published private(set) var taxBasedOnSetting: String = ""
+    @Published private(set) var taxBasedOnSetting: TaxBasedOnSetting?
```
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift]: I think we're not getting this `private(set)` anywhere at the moment, we could also make it `private` and move the `displayString` var to the `TaxBasedOnSetting` enum directly:

```
public enum TaxBasedOnSetting {
    case customerShippingAddress
    case customerBillingAddress
    case shopBaseAddress

    public var displayString: String {
        switch self {
        case .customerBillingAddress:
            return NSLocalizedString("Calculated on billing address",
                                     comment: "The string to show on order taxes when they are calculated based on the billing address")
        case .customerShippingAddress:
            return NSLocalizedString("Calculated on shipping address",
                                     comment: "The string to show on order taxes when they are calculated based on the shipping address")
        case .shopBaseAddress:
            return NSLocalizedString("Calculated on shop base address",
                                     comment: "The string to show on order taxes when they are calculated based on the shop base address")
        }
    }
}
```

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/PaymentSection/OrderPaymentSection.swift [hunk 2]
```diff
@@ -166,7 +176,7 @@ struct OrderPaymentSection: View {
         VStack(alignment: .leading, spacing: Constants.taxesSectionVerticalSpacing) {
             taxSectionTitle
             taxLines
-            if viewModel.taxLineViewModels.isNotEmpty && viewModel.taxBasedOnSetting.isNotEmpty {
+            if viewModel.taxLineViewModels.isNotEmpty && viewModel.taxBasedOnSetting != nil {
```
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Orders/Order Creation/PaymentSection/OrderPaymentSection.swift]: No strong preference but I find the combination of using `.renderedIf()` and declaring the variables more readable for views than `if/else`:
```
    @ViewBuilder private var taxesSection: some View {
        let taxLineNotEmpty = viewModel.taxLineViewModels.isNotEmpty
        let taxBaseNotNil = viewModel.taxBasedOnSetting != nil

        VStack(alignment: .leading, spacing: Constants.taxesSectionVerticalSpacing) {
            taxSectionTitle
            taxLines
            taxBasedOnLine
                .renderedIf(taxLineNotEmpty && taxBaseNotNil)
        }
        .padding(Constants.sectionPadding)
    }
```

**Review by iamgabrielma — APPROVED** [review]
Works well! :shipit:

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift (hunk 1 above)
Comment (toupper) [reply WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift]: Good catch! I made it totally private in 7e1004997d1a0357384ac6cdb833d1005ee763e5 Regarding moving the `displayString` var to the TaxBasedOnSetting enum directly, I prefer leaving it as an extension in the `EditableOrderViewModel`. That way we keep it decoupled with the enum, given that the `displayString` is only related to the `EditableOrderViewModel` screen.

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/PaymentSection/OrderPaymentSection.swift (hunk 2 above)
Comment (toupper) [reply WooCommerce/Classes/ViewRelated/Orders/Order Creation/PaymentSection/OrderPaymentSection.swift]: > No strong preference but I find the combination of using .renderedIf() and declaring the variables more readable for views than if/else:

Me too! Done in 0166b52e581c8bf28dd25a4ede6d817a9ee3feb2

**Pull request merged**